llvm/lib/Target/AIE/AIEHazardRecognizer.cpp

```diff
@@ -563,3 +602,18 @@ AIEHazardRecognizer::getSelectedAltOpcode(MachineInstr *MI) const {
     return It->second;
   return std::nullopt;
 }
+
+MemoryBankBits AIEHazardRecognizer::getMemoryBanks(MachineInstr *MI) const {
+  MemoryBankBits MemoryBankUsed = 0;
+
+  if (!(MI->mayLoad() || MI->mayStore()))
+    return MemoryBankUsed;
+
+  const AIEBaseSubtarget &STI = AIEBaseSubtarget::get(*MI->getMF());
+  for (auto &MMO : MI->memoperands()) {
+    unsigned MemoryBank =
+        STI.getMemoryBanksFromAddressSpace(MMO->getAddrSpace()).to_ulong();
+    MemoryBankUsed |= MemoryBank;
```

Comment:
I see a bitset, an unsigned long, an unsigned (int) and a MemoryBankBits, all designating the same type within two lines. I propose to have a single STI.getMemoryBankBitsFromAddressSpace() to hide the implementation details.